buildkitd/buildkitd.go

```diff
@@ -924,40 +921,44 @@ func makeTLSPath(path string, installationName string) (string, error) {
 }
 
 func addRequiredOpts(settings Settings, installationName string, opts ...client.ClientOpt) ([]client.ClientOpt, error) {
+	server, err := url.Parse(settings.BuildkitAddress)
+	if err != nil {
+		return []client.ClientOpt{}, errors.Wrap(err, "invalid buildkit url")
+	}
+
 	if settings.SatelliteName != "" {
-		return append(opts, client.WithAdditionalMetadataContext(
+		opts = append(opts, client.WithAdditionalMetadataContext(
 			"satellite_name", settings.SatelliteName,
 			"satellite_org", settings.SatelliteOrgID,
 			"satellite_token", settings.SatelliteToken),
-			client.WithCredentials("", "", "", ""), // force buildkit to use a TLS connection
-		), nil
+		)
 	}
 
 	if !settings.UseTCP || !settings.UseTLS {
 		return opts, nil
 	}
 
-	server, err := url.Parse(settings.BuildkitAddress)
-	if err != nil {
-		return []client.ClientOpt{}, errors.Wrap(err, "invalid buildkit url")
-	}
+	if settings.TLSCA == "" && settings.ClientTLSCert == "" && settings.ClientTLSKey == "" {
+		opts = append(opts, client.WithCredentials("", "", "", ""))
```

Comment:
nit: `opts` is never changed past this point, so it's generally more idiomatic to just return inside the `if` statement and remove the `else`.

```go
		return append(opts, client.WithCredentials("", "", "", ""))
	}
	// the rest of the logic
	return opts, nil
```